jax/_src/random.py

```diff
@@ -1246,7 +1246,10 @@ def categorical(key, logits, axis=-1, shape=None):
     _check_shape("categorical", shape, batch_shape)
 
   sample_shape = shape[:len(shape)-len(batch_shape)]
-  return jnp.argmax(gumbel(key, sample_shape + logits.shape, logits.dtype) + logits, axis=axis)
+  return jnp.argmax(
+      gumbel(key, sample_shape + logits.shape, logits.dtype) +
+      logits.reshape((1,) * len(sample_shape) + logits.shape),
```

Comment:
Slight nit: I'd prefer `lax.expand_dims` here, since you are inserting size-1 dimensions. `reshape` is a big hammer and we'd prefer to avoid it where its power is not needed.